mentoring/components/answer.py

```diff
@@ -0,0 +1,235 @@
+        if not student_input and self.default_from:
+            student_input = self.get_model_object(name=self.default_from).student_input
+
+        return student_input
+
+    def mentoring_view(self, context=None):
+        context = context or {}
+        context['self'] = self
+        html = loader.render_template('templates/html/answer_editable.html', context)
+
+        fragment = Fragment(html)
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/answer.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/answer.js'))
+        fragment.initialize_js('AnswerBlock')
+        return fragment
+
+    def submit(self, submission):
+        self.student_input = submission[0]['value'].strip()
+        self.save()
+        log.info(u'Answer submitted for`{}`: "{}"'.format(self.name, self.student_input))
+        return {
+            'student_input': self.student_input,
+            'status': self.status,
+            'weight': self.weight,
+            'score': 1 if self.status == 'correct' else 0,
+        }
+
+    @property
+    def status(self):
+        answer_length_ok = self.student_input
+        if self.min_characters > 0:
+            answer_length_ok = len(self.student_input.strip()) >= self.min_characters
+
+        return 'correct' if answer_length_ok else 'incorrect'
+
+    @property
+    def completed(self):
+        return self.status == 'correct'
+
+    def save(self):
+        """
+        Replicate data changes on the related Django model used for sharing of data accross XBlocks
+        """
+        super(AnswerBlock, self).save()
+
+        student_id = self._get_student_id()
+        if not student_id:
+            return  # save() gets called from the workbench homepage sometimes when there is no student ID
+
+        # Only attempt to locate a model object for this block when the answer has a name
+        if self.name:
+            answer_data = self.get_model_object()
+            if answer_data.student_input != self.student_input:
+                answer_data.student_input = self.student_input
+                answer_data.save()
+
+
+class AnswerRecapBlock(AnswerMixin, StudioEditableXBlockMixin, XBlock):
+    """
+    A block that displays an answer previously entered by the student (read-only).
+    """
+    display_name = String(
+        display_name="Title",
+        help="Title of this answer recap section",
+        scope=Scope.content,
+        default="",
+    )
+    description = String(
+        help="Description of this answer (optional). Can include HTML.",
+        scope=Scope.content,
+        default="",
+        display_name="Description",
+    )
+    editable_fields = ('name', 'display_name', 'description')
+
+    @property
+    def student_input(self):
+        if self.name:
+            return self.get_model_object().student_input
+        return ''
+
+    def fallback_view(self, view_name, context=None):
```

Comment:
Yes, exactly. By defining `fallback_view` like this, it will be used for `mentoring_view`, `student_view`, or any other view that is requested.